tests/unit/unit_test_client.py

```diff
@@ -213,41 +214,208 @@ def _getFileHandleDownload(fileHandleId,  objectId, objectType='FileHandle'):
+        self.patch_getTeam = patch.object(syn, "getTeam", return_value= self.team)
+        self.patch_get_contributors = patch.object(syn, "_get_contributors",
+                                                   return_value=(self.contributors, self.eligibility))
+
+        self.mock_restPOST = self.patch_restPOST.start()
+        self.mock_getEvaluation = self.patch_getEvaluation.start()
+        self.mock_get = self.patch_get.start()
+        self.mock_getTeam = self.patch_getTeam.start()
+        self.mock_get_contributors = self.patch_get_contributors.start()
+
+    def teardown(self):
+        self.patch_restPOST.stop()
+        self.patch_getEvaluation.stop()
+        self.patch_get.stop()
+        self.patch_getTeam.stop()
+        self.patch_get_contributors.stop()
+
+    def test_min_requirements(self):
+        assert_equal(self.submission, syn.submit(self.eval_id, self.entity))
+        expected_submission = self.submission
+        expected_submission.pop('id')
+        expected_submission['teamId'] = None
+        expected_submission['submitterAlias'] = None
+        uri = '/evaluation/submission?etag={0}&submissionEligibilityHash={1}'\
+            .format(self.entity['etag'],self.eligibility['eligibilityStateHash'])
+        self.mock_restPOST.assert_called_once_with(uri, json.dumps(expected_submission))
+        self.mock_get.assert_not_called()
+        self.mock_getTeam.assert_not_called()
+        self.mock_get_contributors.assert_called_once_with(self.eval_id, None)
+
+    def test_only_entity_id_provided(self):
+        assert_equal(self.submission, syn.submit(self.eval_id, self.entity['id']))
+        expected_submission = self.submission
+        expected_submission.pop('id')
+        expected_submission['teamId'] = None
+        expected_submission['submitterAlias'] = None
+        uri = '/evaluation/submission?etag={0}&submissionEligibilityHash={1}' \
+            .format(self.entity['etag'], self.eligibility['eligibilityStateHash'])
+        self.mock_restPOST.assert_called_once_with(uri, json.dumps(expected_submission))
+        self.mock_get.assert_called_once_with(self.entity['id'], downloadFile=False)
+        self.mock_getTeam.assert_not_called()
+        self.mock_get_contributors.assert_called_once_with(self.eval_id, None)
+
+    def test_team_is_given(self):
+        assert_equal(self.submission, syn.submit(self.eval_id, self.entity, team=self.team['id']))
+        expected_submission = self.submission
+        expected_submission.pop('id')
+        uri = '/evaluation/submission?etag={0}&submissionEligibilityHash={1}' \
+            .format(self.entity['etag'], self.eligibility['eligibilityStateHash'])
+        self.mock_restPOST.assert_called_once_with(uri, json.dumps(expected_submission))
+        self.mock_get.assert_not_called()
+        self.mock_getTeam.assert_called_once_with(self.team['id'])
+        self.mock_get_contributors.assert_called_once_with(self.eval_id, self.team)
+
```

Comment:
add test where submissionEligibilityHash is not a part of the uri

Reply:
also consider refactoring the restPOST call including the cunstructiong of the uri based on submissionEligibilityHash into its own method.